week08/movie/src/pages/HomePage.jsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import styled from "styled-components";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import Movie from "../components/Movie";
+import { BeatLoader } from "react-spinners";
+
+const Container = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+`;
+
+const Loader = styled(BeatLoader)`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 100vh;
+`;
+
+const SortContainer = styled.div`
+    width: 100vw;
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+`;
+
+const SortList = styled.div`
+    margin-top: 10px;
+    margin-right: 30px;
+`;
+
+const MovieList = styled.div`
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+`;
+
+const StyledLink = styled(Link)`
+    text-decoration: none;
+    color: inherit;
+`;
+
+const Text = styled.div`
+    font-size: 40px;
+    margin-left: 15px;
+    margin-top: 5px;
+`;
+
+const HomePage = () => {
+    const [loading, setLoading] = useState(true);
+    const [movies, setMovies] = useState([]);
+    const [sort, setSort] = useState("download_count");
```

Comment:
3. 컴포넌트가 렌더링되어 변수 `sort`의 값이 초기화될 때, localStorage에서 key가 `SAVED_SORT_KEY`인 데이터가 존재한다면, 변수 `sort`의 초기값을 그 데이터로 설정합니다. 존재하지 않는다면, 변수 `sort`의 초기값을 `"download_count"`로 설정합니다.

```suggestion
    const [sort, setSort] = useState(() => {
        const savedSort = localStorage.getItem(SAVED_SORT_KEY);
        return savedSort ? JSON.parse(savedSort) : "download_count";
    });
```